backend/tessend/tessend.go

```diff
@@ -200,20 +217,44 @@ func (t *Tessend) startMessageHandler() {
 		}
 
 		tessen, ok := msg.(*corev2.TessenConfig)
-		if !ok {
-			logger.WithField("msg", msg).Errorf("received non-TessenConfig on tessen message channel")
-			return
+		if ok {
+			data := t.getDataPayload()
+			t.getTessenConfigMetrics(time.Now().UTC().Unix(), tessen, data)
+			logger.WithFields(logrus.Fields{
+				"url":                       t.url,
+				"id":                        data.Cluster.ID,
+				"opt-out":                   tessen.OptOut,
+				data.Metrics.Points[0].Name: data.Metrics.Points[0].Value,
+			}).Info("sending opt-out status event to tessen")
+			_ = t.send(data)
+			continue
 		}
 
-		data := t.getDataPayload()
-		t.getTessenConfigMetrics(time.Now().UTC().Unix(), tessen, data)
-		logger.WithFields(logrus.Fields{
-			"url":                       t.url,
-			"id":                        data.Cluster.ID,
-			"opt-out":                   tessen.OptOut,
-			data.Metrics.Points[0].Name: data.Metrics.Points[0].Value,
-		}).Info("sending opt-out status event to tessen")
-		_ = t.send(data)
+		metrics, ok := msg.([]corev2.MetricPoint)
+		if ok {
+			if t.enabled() {
+				data := t.getDataPayload()
+				now := time.Now().UTC().Unix()
```

Comment:
Just so you're aware, this doesn't compute the offset to UTC from the local TZ - `time.Now().Unix()` delivers the same result as `time.Now().UTC().Unix()`.

https://golang.org/pkg/time/#Time.UTC

Reply:
![giphy](https://user-images.githubusercontent.com/10634078/59066836-cf726a80-8864-11e9-921b-95edfbfa4f8e.gif)

Reply:
Ah well, that answers my comment a few lines above!